modules/unsupported/jdbc-ng/jdbc-hana/src/main/java/org/geotools/data/hana/HanaDialect.java

```diff
@@ -0,0 +1,892 @@
+import org.opengis.referencing.crs.GeographicCRS;
+import org.opengis.referencing.crs.SingleCRS;
+
+/**
+ * A prepared statement SQL dialect for SAP HANA.
+ *
+ * @author Stefan Uhrig, SAP SE
+ */
+public class HanaDialect extends PreparedStatementSQLDialect {
+
+    /** UUID suffix for auto-generated identifiers for avoiding name clashes */
+    private static final String HANA_UUID = "8E468249703240F0ACDE78162124A62F";
+
+    private static final String SEQUENCE_SUFFIX = "_SEQ_" + HANA_UUID;
+
+    private static final String METADATA_TABLE_NAME = "METADATA_" + HANA_UUID;
+
+    private static final Map<String, Class<?>> TYPE_NAME_TO_CLASS = new HashMap<String, Class<?>>();
+
+    static {
+        TYPE_NAME_TO_CLASS.put("TINYINT", Short.class);
+        TYPE_NAME_TO_CLASS.put("SMALLINT", Short.class);
+        TYPE_NAME_TO_CLASS.put("INTEGER", Integer.class);
+        TYPE_NAME_TO_CLASS.put("BIGINT", Long.class);
+        TYPE_NAME_TO_CLASS.put("DECIMAL", BigDecimal.class);
+        TYPE_NAME_TO_CLASS.put("REAL", Float.class);
+        TYPE_NAME_TO_CLASS.put("DOUBLE", Double.class);
+        TYPE_NAME_TO_CLASS.put("CHAR", String.class);
+        TYPE_NAME_TO_CLASS.put("VARCHAR", String.class);
+        TYPE_NAME_TO_CLASS.put("BINARY", byte[].class);
+        TYPE_NAME_TO_CLASS.put("VARBINARY", byte[].class);
+        TYPE_NAME_TO_CLASS.put("DATE", Date.class);
+        TYPE_NAME_TO_CLASS.put("TIME", Time.class);
+        TYPE_NAME_TO_CLASS.put("TIMESTAMP", Timestamp.class);
+        TYPE_NAME_TO_CLASS.put("CLOB", String.class);
+        TYPE_NAME_TO_CLASS.put("BLOB", byte[].class);
+        TYPE_NAME_TO_CLASS.put("NCHAR", String.class);
+        TYPE_NAME_TO_CLASS.put("NVARCHAR", String.class);
+        TYPE_NAME_TO_CLASS.put("ALPHANUM", String.class);
+        TYPE_NAME_TO_CLASS.put("NCLOB", String.class);
+        TYPE_NAME_TO_CLASS.put("SMALLDECIMAL", BigDecimal.class);
+        TYPE_NAME_TO_CLASS.put("TEXT", String.class);
+        TYPE_NAME_TO_CLASS.put("BINTEXT", byte[].class);
+        TYPE_NAME_TO_CLASS.put("SHORTTEXT", String.class);
+        TYPE_NAME_TO_CLASS.put("SECONDDATE", Timestamp.class);
+        TYPE_NAME_TO_CLASS.put("ST_POINT", Point.class);
+        TYPE_NAME_TO_CLASS.put("ST_GEOMETRY", Geometry.class);
+        TYPE_NAME_TO_CLASS.put("BOOLEAN", Boolean.class);
+    };
+
+    private static final Map<Integer, Class<?>> SQL_TYPE_TO_CLASS =
+            new HashMap<Integer, Class<?>>();
+
+    static {
+        SQL_TYPE_TO_CLASS.put(-4, byte[].class); // BLOB
+        SQL_TYPE_TO_CLASS.put(-8, String.class); // NCHAR
+        SQL_TYPE_TO_CLASS.put(-10, String.class); // NCLOB
+        SQL_TYPE_TO_CLASS.put(3000, BigDecimal.class); // SMALLDECIMAL
+        SQL_TYPE_TO_CLASS.put(29812, Geometry.class); // ST_GEOMETRY
+    }
+
+    private static final Map<Class<?>, Integer> CLASS_TO_SQL_TYPE =
+            new HashMap<Class<?>, Integer>();
+
+    static {
+        CLASS_TO_SQL_TYPE.put(Geometry.class, 29812);
```

Comment:
The 29812 "magic number" is used in several places, I'd use a constant.

Reply:
I've introduced a constant for the type code.